src/weierstrass/curve.rs

```diff
@@ -76,6 +77,52 @@ impl EllipticCurve for Curve {
             z: self.f.one(),
         })
     }
+    fn deserialize(&self, buf: &[u8]) -> Result<Self::Point,Error> {
+        let p = self.f.get_modulus();
+        let max_bytes = (p.bits()+7)/8;
+        if buf.len() == 0 {
+            return Err(Error::new(ErrorKind::Other, "Input buffer is empty."));
+        }
+        let tag = buf[0];
+        match tag {
+            0x04 => {
+                if buf.len() != 2*max_bytes+1 {
+                    return Err(Error::new(ErrorKind::Other, "Invalid bytes for deserialization"));
+                }
+                let x = self.f.elt(BigInt::from_bytes_be(Sign::Plus, &buf[1..max_bytes+1]));
```

Comment:
Before passing the integer to `elt`, it must be verified that the integer is in [0,p-1], otherwise return error. This verification is part of the deserialization method.